### PR TITLE
[9.0][MIG] Migrate help_online to 9.0

### PR DESCRIPTION
**Pull request by lmignon**

### CODE REVIEW

**Comment by lmignon** [comment]
WIP: WAITING FOR RUNBOT TO TEST THE MODULE ON 9.0c

**Comment by lmignon** [comment]
Tested on Odoo 9c et 9e

File: help_online/README.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/162/9.0
+
+
+Known issues / Roadmap
+======================
+
+Even if the generated  urls to the documentation contains an anchor (website/hel-xx#view_type),
+it's no more possible to insert/edit anchors elemetns into the website since this functionnality is not supported
```
Comment (elicoidal) [inline help_online/README.rst]: s/elemetns/elements

File: help_online/README.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+------------
+
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* Jonathan Nemry <jonathan.nemry@acsone.eu>
+* Cédric Pigeon <cedric.pigeon@acsone.eu>
+
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
```
Comment (elicoidal) [inline help_online/README.rst]: http=>https (x3)

File: help_online/views/website_help_online.xml [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+    <data>
 
-<template id="assets_editor" inherit_id="website.assets_editor" name="Help online customization">
-    <xpath expr="." position="inside">
-        <script type="text/javascript" src="/help_online/static/src/js/website_help_online.editor.js" groups="base.group_website_publisher"></script>
-    </xpath>
-</template>
+        <!-- TODO IN 9.0 CKEditor is replaced by summernote. This one doesn't provide
+        a way to insert/edit anchor elements. -->  
+        <!-- template id="assets_editor" inherit_id="website.assets_editor"
```
Comment (elicoidal) [inline help_online/views/website_help_online.xml]: please remove dead code

**Comment by lmignon** [comment]
@elicoidal Thank you for the review. Changes done and test coverage improved.

File: help_online/security/help_online_groups.xml [hunk 4]
```diff
@@ -1,4 +1,5 @@
-<openerp>
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
     <data>
```
Comment (lasley) [inline help_online/security/help_online_groups.xml]: No need for the `data` tag w/ the new `odoo` tags

File: help_online/security/help_online_rules.xml [hunk 5]
```diff
@@ -1,32 +1,32 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
```
Comment (lasley) [inline help_online/security/help_online_rules.xml]: `<odoo noupdate="1">`

Comment (lasley) [inline help_online/security/help_online_rules.xml]: `<odoo noupdate="1">`

File: help_online/views/help_online_view.xml [hunk 6]
```diff
@@ -1,7 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- vim:fdn=3:
--->
-<openerp>
+<odoo>
     <data>
```
Comment (lasley) [inline help_online/views/help_online_view.xml]: No need for data tag

File: help_online/views/ir_ui_view_view.xml [hunk 7]
```diff
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
     <data noupdate="0">
```
Comment (lasley) [inline help_online/views/ir_ui_view_view.xml]: No need for data tag

File: help_online/wizards/export_help_wizard_view.xml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
     <data>
```
Comment (lasley) [inline help_online/wizards/export_help_wizard_view.xml]: No need for data tag

**Comment by lasley** [comment]
Rebuilding Runbot for functional, code is 👍  outside of some incredibly minor comments.

Also pardon my pun, but that XML and routing work in the wizard was... wizardry 😆

File: help_online/wizards/export_help_wizard.py [hunk 9]
```diff
@@ -43,94 +30,114 @@ class ExportHelpWizard(models.TransientModel):
...
+        Rule('/website/image/<xmlid>/<field>'),
+        Rule('/website/image/<xmlid>/<field>/<int:width>x<int:height>'),
+        Rule('/website/image/<model>/<id>/<field>'),
+        Rule('/website/image/<model>/<id>/<field>/<int:width>x<int:height>')
+    ])
+
+    def _manage_images_on_page(self, page_node, data_node, exported_resources):
         """
-            - Extract images from page and generate a xml node
+            - Extract images from page and generate an node
```
Comment (tedsalmon) [inline help_online/wizards/export_help_wizard.py]: Nit picking - I believe you meant for the doc string to read:

> - Extract images from page and generate an xml node

**Comment by lasley** [comment]
Functional as well 🎉

**Comment by lmignon** [comment]
@lasley Thank you for the review. Changes done.

**Comment by lasley** [comment]
Changes 👍  thanks @lmignon

**Review by lasley — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by JonathanNEMRY — APPROVED** [review]

**Comment by lmignon** [comment]
@lasley IMO this one is ready to merge 3 :+1: and a proper git history

**Pull request merged**